subprojects/core/src/main/java/org/gradle/caching/internal/BuildCacheServiceProvider.java

```diff
@@ -140,4 +148,32 @@ public String getRole() {
             return role;
         }
     }
+
+    private class FinalizeBuildCacheConfigurationBuildOperation implements RunnableBuildOperation {
+        private final BuildCache local;
+        private final BuildCache remote;
+
+        private FinalizeBuildCacheConfigurationBuildOperation(BuildCache local, BuildCache remote) {
+            this.local = local;
+            this.remote = remote;
+        }
+
+        @Override
+        public void run(BuildOperationContext buildOperationContext) {
+            boolean enabled = startParameter.isBuildCacheEnabled();
+            buildOperationContext.setResult(new FinalizeBuildCacheConfigurationDetails.Result(enabled, convertToWrapper(local, enabled), convertToWrapper(remote, enabled)));
+        }
+
+        @Nullable
+        private BuildCacheWrapper convertToWrapper(BuildCache buildCache, boolean enabled) {
+            return buildCache == null || !enabled
```

Comment:
Not sure if I follow your concern, we want to return a null wrapper if the original BuildCache is null or the overall build cache concept is disables, that's why that check is needed. I don't see the redundancy, sorry.